### PR TITLE
title_format plot option now includes dimensions

### PR DESCRIPTION
**Pull request by philippjfr**
As the title says this allows more customizing the plot title_format as suggested in #425.

### CODE REVIEW

**Comment by philippjfr** [comment]
The ``title_format`` plot option now allows enabling or disabling dimensions in the title. The only way I was able to do so without losing flexibility was to let the title accept a separator variable: "{label} {group}{separator}{dimensions}". Since bokeh doesn't support newline characters I'm tempted to remove this again and simply set different default title_format, for matplotlib I'd have "{label} {group}\n{dimensions}" and for bokeh I'd have "{label} {group} - {dimensions}".

File: holoviews/plotting/mpl/plot.py [hunk 1]
```diff
@@ -916,7 +916,7 @@ def _compute_gridspec(self, layout):
 
         # Create title handle
         if self.show_title and len(self.coords) > 1:
-            title = self.handles['fig'].suptitle('', **self._fontsize('title'))
+            title = self.handles['fig'].suptitle('', y=1.05, **self._fontsize('title'))
```
Comment (jlstevens) [inline holoviews/plotting/mpl/plot.py]: Is this a magic number? I assume this makes it look better.

Comment (philippjfr) [inline holoviews/plotting/mpl/plot.py]: Yea, generally avoids overlap between the plot title and the overall layout title. Not sure how best to provide control over it.

File: holoviews/plotting/plot.py [hunk 2]
```diff
@@ -145,8 +145,9 @@ class DimensionedPlot(Plot):
     show_title = param.Boolean(default=True, doc="""
         Whether to display the plot title.""")
 
-    title_format = param.String(default="{label} {group}", doc="""
-        The formatting string for the title of this plot.""")
+    title_format = param.String(default="{label} {group}\n{dimensions}", doc="""
+        The formatting string for the title of this plot, allows defining
+        a label group separator and dimension labels.""")
```
Comment (jlstevens) [inline holoviews/plotting/plot.py]: Definitely an improvement as long as the old tests pass (i.e backwards compatible).

Comment (philippjfr) [inline holoviews/plotting/plot.py]: It probably won't be, there may be fewer whitespace now and things may have shifted around a little bit.

Comment (jlstevens) [inline holoviews/plotting/plot.py]: Ok. That means there will be quite a lot of test data to update. We need to be careful about transient glitches and need to review the display output thoroughly.

**Comment by philippjfr** [comment]
Looks good for merging, the display changes are the deletion of trailing whitespace in the titles when no label is supplied.

**Comment by jlstevens** [comment]
Ok. Tests are passing. Merging.

**Pull request merged**